lib/chef/provider/package/cab.rb

```diff
@@ -74,9 +74,13 @@ def remove_package(name, version)
         end
 
         def dism_command(command)
-          shellout = Mixlib::ShellOut.new("dism.exe /Online /English #{command} /NoRestart", timeout: new_resource.timeout)
           with_os_architecture(nil) do
-            shellout.run_command
+            begin
+              shell_out!("dism.exe /Online /English #{command} /NoRestart", { timeout: new_resource.timeout })
+            rescue Exception => e
```

Comment:
Can we figure out exactly what we need to rescue so we're not rescuing everything. That has some nasty side effects.

Reply:
More specifically it should probably use `shell_out` without the bang and then based on the exit code either force the shell_out object to raise or to do the custom error handling.

I know we do this somewhere already.